File: propet_f/lib/model/Goods.dart
class Goods{

  int gidx;
  String gname;
  String gexplan;
  String gingre;
  int gprice;
  String glink;
  String gsoldout;

  Goods({
    required this.gidx,
    required this.gname,
    required this.gexplan,
    required this.gingre,
    required this.gprice,
    required this.glink,
    required this.gsoldout,

  });

  // JSON을 Users 객체로 변환하는 factory 생성자
  factory Goods.fromJson(Map<String, dynamic> json) {
    return Goods(
      gidx: json['gidx'],
      gname: json['gname'],
      gexplan: json['gexplan'],
      gingre: json['gingre'],
      gprice: json['gprice'],
      glink: json['glink'],
      gsoldout: json['gsoldout'],
    );
  }

  // Object -> json으로 바꾸는 형태
  Map<String, dynamic> toJson() => {
    'gidx': gidx,
    'gname': gname,
    'gexplan': gexplan,
    'gingre': gingre,
    'gprice': gprice,
    'glink': glink,
    'gsoldout': gsoldout,
  };


}

File: propet_c/Propet/src/main/java/com/smhrd/flutter/model/Goods.java
package com.smhrd.flutter.model;

import java.math.BigDecimal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.Setter;

@Entity(name="goods")
@Getter
@Setter
public class Goods {

	@Id // PK 지정 long 으로 자주 사용
	@GeneratedValue(strategy = GenerationType.IDENTITY) // 고유키 자동생성
	@Column(name = "g_idx")
	private long gidx;
	
	@Column(name ="g_name", length = 50) //상품이름
	private String gname;
	
	@Column(name ="g_explan", length = 50) //설명 - 오타잇음
	private String gexplan;
	
	
	@Column(name ="g_ingre", length = 50) // 성분
	private String gingre;
	
	@Column(name ="g_price") //가격
	private long gprice;
	
	@Column(name ="g_link", length = 100) //중성화여부
	private String glink;
	
	@Column(name ="g_soldout", length = 1) //질환여부
	private String gsoldout;
	
}
